v22.1/row-level-ttl.md

```diff
@@ -0,0 +1,323 @@
+
+## Syntax overview
+
+TTLs are defined on a per-table basis using SQL statements. The syntax for creating a table with an automatically managed TTL extends the [`storage_parameter` syntax](sql-grammar.html#opt_with_storage_parameter_list). For example, the SQL statement
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE TABLE ttl_test (
+  id UUID PRIMARY KEY default gen_random_uuid(),
+  description TEXT,
+  inserted_at TIMESTAMP default current_timestamp()
+) WITH (ttl_expire_after = '3 minutes');
+~~~
+
+has the following effects:
+
+<a name="crdb-internal-expiration"></a>
+
+1. Creates a repeating [scheduled job](#view-ttl-jobs) for the `events` table.
+2. Adds a hidden column `crdb_internal_expiration` of type [`TIMESTAMPTZ`](timestamp.html) to represent the TTL.
+3. Implicitly adds the `ttl` and `ttl_automatic_column` [storage parameters](#ttl-storage-parameters).
+
+To see the hidden column and the storage parameters on the `events` table, enter the [`SHOW CREATE TABLE`](show-create.html) statement:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SHOW CREATE TABLE ttl_test;
+~~~
+
+~~~
+  table_name |                                                                                           create_statement
+-------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+  ttl_test   | CREATE TABLE public.ttl_test (
+             |     id UUID NOT NULL DEFAULT gen_random_uuid(),
+             |     description STRING NULL,
+             |     inserted_at TIMESTAMP NULL DEFAULT current_timestamp():::TIMESTAMP,
+             |     crdb_internal_expiration TIMESTAMPTZ NOT VISIBLE NOT NULL DEFAULT current_timestamp():::TIMESTAMPTZ + '00:03:00':::INTERVAL ON UPDATE current_timestamp():::TIMESTAMPTZ + '00:03:00':::INTERVAL,
+             |     CONSTRAINT ttl_test_pkey PRIMARY KEY (id ASC)
+             | ) WITH (ttl = 'on', ttl_automatic_column = 'on', ttl_expire_after = '00:03:00':::INTERVAL)
+(1 row)
+~~~
+
+## TTL storage parameters
+
+The settings that control the behavior of Row-Level TTL are provided using [storage parameters](sql-grammar.html#opt_with_storage_parameter_list). These parameters can be set during table creation using [`CREATE TABLE`](#create-a-table-with-row-level-ttl), added to an existing table using the [`ALTER TABLE`](#add-row-level-ttl-to-an-existing-table) statement, or [reset to default values](#reset-a-storage-parameter-to-its-default-value).
+
+| Description                                              | Option                                                                                                                                                                       |
```

Comment:
some of these settings (as you say) have a default of the cluster setting, instead of the defaults mentioned here. maybe worth adding a column to signify that.